subsys/usb/usb_c/usbc_tc_common.c

```diff
@@ -55,8 +56,15 @@ void tc_run(const struct device *dev, const int32_t dpm_request)
 	/* fall through */
 	case SM_INIT:
 		/* Initialize the Type-C layer */
-		if (tc_init(dev) < 0) {
-			LOG_ERR("Error initializing Type-C layer");
+		ret = tc_init(dev);
+		if (ret) {
+			if (ret != -EAGAIN) {
+				/* Transition to Disabled State */
+				LOG_ERR("Disabling the Type-C Layer");
+				data->tc_enabled = false;
+				tc_set_state(dev, TC_DISABLED_STATE);
+			}
+
 			break;
 		}
```

Comment:
```suggestion
		if (ret != -EAGAIN) {
			/* Transition to Disabled State */
			LOG_ERR("Disabling the Type-C Layer");
			data->tc_enabled = false;
			tc_set_state(dev, TC_DISABLED_STATE);
		}

		if (ret != 0) {
			break;
		}
```

Reply:
It will require double checking for `!= 0`, but I did that for you to compare